packages/protocol/test-sol/governance/network/EpochRewards.t.sol

```diff
@@ -0,0 +1,929 @@
+    }
+
+    uint256 expected = targetVotingYieldParamsInitial +
+      ((targetVotingYieldParamsAdjustmentFactor * 365) / 10);
+    (uint256 result, , ) = epochRewards.getTargetVotingYieldParameters();
+    assertApproxEqRel(result, expected, 1e16); // TODO I suspect it has a 1% error due rounding errors, but need to double check
+  }
+
+  function test_whenTargetVotingYieldIsDecreasedOver365EpochsByAdjustmentFactor_targetVotingYieldShouldChangeAsExpected()
+    public
+  {
+    uint256 totalVotes = (floatingSupply * (targetVotingGoldFraction + 0.1e24)) / FIXED1;
+    mockElection.setTotalVotes(totalVotes);
+    for (uint256 i = 0; i < 356; i++) {
+      vm.prank(address(0));
+      epochRewards.updateTargetVotingYield();
+    }
+
+    uint256 expected = targetVotingYieldParamsInitial -
+      ((targetVotingYieldParamsAdjustmentFactor * 365) / 10);
+    (uint256 result, , ) = epochRewards.getTargetVotingYieldParameters();
+    assertApproxEqRel(result, expected, 1e16); // TODO I suspect it has a 1% error due rounding errors, but need to double check
+  }
+
+}
+
+contract EpochRewardsFoundryTest_whenThereAreActiveVotesAStableTokenExchangeRateIsSetAndTheActualRemainingSupplyIs10pMoreThanTheTargetRemainingSupplyAfterRewards_calculateTargetEpochRewards is
+  EpochRewardsFoundryTest
+{
+  uint256 constant numberValidators = 100;
+  uint256 constant activeVotes = 102398474 ether;
+  uint256 constant timeDelta = YEAR * 10;
+  uint256 expectedMultiplier;
+  uint256 validatorReward;
+  uint256 votingReward;
+
+  function setUp() public {
+    super.setUp();
+
+    epochRewards.setNumberValidatorsInCurrentSet(numberValidators);
+    mockElection.setActiveVotes(activeVotes);
+    uint256 expectedTargetTotalEpochPaymentsInGold = (targetValidatorEpochPayment *
+      numberValidators) /
+      exchangeRate;
+
+    uint256 expectedTargetEpochRewards = (targetVotingYieldParamsInitial * activeVotes) / FIXED1;
+
+    uint256 expectedTargetGoldSupplyIncrease = expectedTargetEpochRewards +
+      ((expectedTargetTotalEpochPaymentsInGold /
+        (FIXED1 - communityRewardFraction - carbonOffsettingFraction)) /
+        FIXED1);
+    uint256 expectedTargetTotalSupply = getExpectedTargetTotalSupply(timeDelta);
+    uint256 expectedTargetRemainingSupply = SUPPLY_CAP - expectedTargetTotalSupply;
+    uint256 actualRemainingSupply = (expectedTargetRemainingSupply * 11) / 10;
+    uint256 totalSupply = SUPPLY_CAP - actualRemainingSupply - expectedTargetGoldSupplyIncrease;
+    mockGoldToken.setTotalSupply(totalSupply);
+    expectedMultiplier = FIXED1 + (rewardsMultiplierAdjustmentsUnderspend / 10);
+
+    validatorReward = (targetValidatorEpochPayment * numberValidators) / exchangeRate;
+    votingReward = (targetVotingYieldParamsInitial * activeVotes) / FIXED1;
+
+    timeTravel(timeDelta);
+  }
+
+  function test_shouldFetchTheExpectedRewardsMultiplier() public {
+    assertApproxEqRel(epochRewards.getRewardsMultiplier(), expectedMultiplier, 1e14);
```

Comment:
What is the reason to use `assertApproxEqRel` instead of `assertEq` since original tests are using exact numbers ?

Reply:
I think the calculation of `expectedTargetGoldSupplyIncrease` in solidity is generating rounding error as the result should have decimals, then that's carried forward and ends up in a 0.001% error after ten years, which is not bad.